amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/MultiStreamLease.java

```diff
@@ -0,0 +1,91 @@
+package software.amazon.kinesis.leases;
+
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+import lombok.Setter;
+import lombok.experimental.Accessors;
+import org.apache.commons.lang3.Validate;
+
+import java.util.Objects;
+
+import static com.google.common.base.Verify.verifyNotNull;
+
+@Setter
+@NoArgsConstructor
+@Getter
+@Accessors(fluent = true)
+public class MultiStreamLease extends Lease {
```

Comment:
I believe any lease class we introduce will always be a subset of the subsequent lease. For instance Lease will be a subset of MultiStreamLease, MultiStreamLease will be a subset of MultiStreamMultiAccountLease, MultiStreamMultiAccountLease will be a subset of MultiStreamMultiAccountTwoPhaseCommitLease. The problem will be when there is a need for MultiStream + twoPhase commit only and without multiaccount support, or when there is a need for say basic lease + two phase commit. We can use strategy pattern, if we plan to introduce 2-phase commit as an independent feature of multistream/multiaccount lease or if we decide that leases can be combinational rather than incremental.